### PR TITLE
bump version to 2.6.1

### PR DESCRIPTION
**Pull request by mnitchev**
The gist if the changes are:
* Bump the image version to the new one. I've also changed the image name. This is because the images from upstream are now in a different registry and the name also changed.
* Keep all the giantswarm changes to the chart. I think I managed to keep everything but please take a look if anything is deleted/modified.
* Use vendir version upstream chart
* Fix the schema to include new values.

Everything else is just pulled changes from the upstream chart.

### CODE REVIEW

**Comment by mnitchev** [comment]
I also deployed this and it seems to be running fine, but I have no idea how to test this. Maybe @giantswarm/team-cabbage can help.

**Comment by ubergesundheit** [comment]
On a WC:

- Create a Service of type LoadBalancer with annotations `service.beta.kubernetes.io/aws-load-balancer-type: external` and `service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: instance`. This should be enough for the in-tree aws lb controller to leave the service alone and let the aws-load-balancer-controller take over
- Install the previous version
- Try to upgrade to your new version
- Change something on your Service. Pick any annotation from [this page](https://kubernetes-sigs.github.io/aws-load-balancer-controller/v2.6/guide/service/annotations/) and check if the controller does it

**Comment by kopiczko** [comment]
You can install IC like this:

```
values=(
  --set "baseDomain=pawe3.gaws.gigantic.io"
  --set "controller.allowSnippetAnnotations=true"
  --set "controller.resources.requests.cpu=1"
  --set "controller.resources.requests.memory=500Mi"
  --set "provider=capa"
  --set-string "controller.config.use-proxy-protocol=false"
  --set "controller.service.loadBalancerClass=service.k8s.aws/nlb"
  --set-json 'controller.service.annotations={"service.beta.kubernetes.io/aws-load-balancer-scheme":"internet-facing","service.beta.kubernetes.io/aws-load-balancer-ip-address-type":"dualstack"}'
)
helm --kube-context t-grizzly-pawe3 upgrade --install \
  "${values[@]}" \
  -n kube-system \
  --version 3.4.2 \
  ingress-nginx oci://giantswarmpublic.azurecr.io/giantswarm-catalog/ingress-nginx
```

Check if it produced NLB for the ingress service and you can install our https://github.com/giantswarm/helloworld manifest to test ingress.

**Comment by mnitchev** [comment]
hmm getting a lot of these errors
```
{"level":"error","ts":"2024-01-25T16:09:00Z","msg":"Reconciler error","controller":"service","object":{"name":"ingress-nginx-controller","namespace":"kube-system"},"namespace":"kube-system","name":"ingress-nginx-controller","reconcileID":"836f2072-68dd-4716-848a-ea2e75940576","error":"unable to resolve at least one subnet (3 match VPC and tags, 3 tagged for other cluster)"}
{"level":"error","ts":"2024-01-25T16:09:00Z","msg":"Reconciler error","controller":"service","object":{"name":"ingress-nginx-controller","namespace":"kube-system"},"namespace":"kube-system","name":"ingress-nginx-controller","reconcileID":"f48ad55c-84ab-43f8-80f3-49b1e79fb745","error":"unable to resolve at least one subnet (3 match VPC and tags, 3 tagged for other cluster)"}
{"level":"error","ts":"2024-01-25T16:09:00Z","msg":"Reconciler error","controller":"service","object":{"name":"ingress-nginx-controller","namespace":"kube-system"},"namespace":"kube-system","name":"ingress-nginx-controller","reconcileID":"8fe8fb3b-8ea9-49e8-be7e-98596fc0adae","error":"unable to resolve at least one subnet (3 match VPC and tags, 3 tagged for other cluster)"}
{"level":"error","ts":"2024-01-25T16:09:01Z","msg":"Reconciler error","controller":"service","object":{"name":"ingress-nginx-controller","namespace":"kube-system"},"namespace":"kube-system","name":"ingress-nginx-controller","reconcileID":"92c6781e-b6ef-45e4-ab4e-e83f953c3fd1","error":"unable to resolve at least one subnet (3 match VPC and tags, 3 tagged for other cluster)"}
{"level":"error","ts":"2024-01-25T16:09:03Z","msg":"Reconciler error","controller":"service","object":{"name":"ingress-nginx-controller","namespace":"kube-system"},"namespace":"kube-system","name":"ingress-nginx-controller","reconcileID":"3aef85d0-95b4-4c56-8d6e-087b01cb21fe","error":"unable to resolve at least one subnet (3 match VPC and tags, 3 tagged for other cluster)"}
{"level":"error","ts":"2024-01-25T16:09:05Z","msg":"Reconciler error","controller":"service","object":{"name":"ingress-nginx-controller","namespace":"kube-system"},"namespace":"kube-system","name":"ingress-nginx-controller","reconcileID":"c597af84-0e0a-4dce-a153-49a06a2bdf53","error":"unable to resolve at least one subnet (3 match VPC and tags, 3 tagged for other cluster)"}
```
Going to investigate on Monday

**Comment by mnitchev** [comment]
I'm also trying it on vintage since that's what I assumed this was being used on

**Review by whites11 — APPROVED** [review]
seems legit

**Comment by mnitchev** [comment]
With some small adjustments I got it working. I tested it on both vintage and capa and it seems to work with setting `service.beta.kubernetes.io/load-balancer-source-ranges` on the LoadBalancer Service. One note though: it's only possible to add that option when creating the LoadBalancer, since you can only create an LB with or without or without a Security Group and you can't add/remove one after the fact. So if you need that option you also have to recreate it.
Unless there's anything else you want me to check @giantswarm/team-cabbage I'll merge and release this.

File: helm/aws-load-balancer-controller/templates/deployment.yaml [hunk 1]
```diff
@@ -63,11 +69,18 @@ spec:
         {{- if .Values.ingressClass }}
         - --ingress-class={{ .Values.ingressClass }}
         {{- end }}
-        {{- include "aws-load-balancer-controller.regionArg" . | nindent 8 }}
-        {{- include "aws-load-balancer-controller.vpcIdArg" . | nindent 8 }}
```
Comment (fiunchinho) [inline helm/aws-load-balancer-controller/templates/deployment.yaml:67]: Was this change intentional? @mnitchev 

The `aws.region` and `aws.vpcID` values are only present on vintage in the `cluster-values` configmap, but they are not in the CAPI `cluster-values configmap.

I'm not sure how to make it work for both vintage and CAPI.
